model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -846,65 +868,190 @@ async def create_vllm_bundle(
-                        command=command,
-                        streaming_command=command,
-                        protocol="http",
-                        readiness_initial_delay_seconds=10,
-                        healthcheck_route="/health",
-                        predict_route="/predict",
-                        streaming_predict_route="/stream",
-                        extra_routes=[
-                            OPENAI_CHAT_COMPLETION_PATH,
-                            OPENAI_COMPLETION_PATH,
-                        ],
-                        env={},
-                    ),
-                    metadata={},
-                ),
+                create_model_bundle_v2_request,
+                do_auth_check=False,
+                # Skip auth check because llm create endpoint is called as the user itself,
+                # but the user isn't directly making the action. It should come from the fine tune
+                # job.
+            )
+        ).model_bundle_id
+
+    async def create_vllm_multinode_bundle(
+        self,
+        user: User,
+        model_name: str,
+        framework_image_tag: str,
+        endpoint_unique_name: str,
+        num_shards: int,
+        nodes_per_worker: int,
+        quantize: Optional[Quantization],
+        checkpoint_path: Optional[str],
+        chat_template_override: Optional[str],
+    ):
+        leader_command = self._create_vllm_bundle_command(
+            model_name,
+            framework_image_tag,
+            num_shards,
+            quantize,
+            checkpoint_path,
+            chat_template_override,
+            multinode=True,
+            is_worker=False,
+            nodes_per_worker=nodes_per_worker,
+        )
+        worker_command = self._create_vllm_bundle_command(
+            model_name,
+            framework_image_tag,
+            num_shards,
+            quantize,
+            checkpoint_path,
+            chat_template_override,
+            multinode=True,
+            is_worker=True,
+            nodes_per_worker=nodes_per_worker,
+        )
+
+        # These env vars e.g. K8S_OWN_POD_NAME, K8S_OWN_POD_NAME, K8S_OWN_NAMESPACE, K8S_LWS_CLUSTER_SIZE will be filled in automatically for all LWS pods through
+        # Launch's k8s_endpoint_resource_delegate
+        common_vllm_envs = {
+            "VLLM_HOST_IP": "$(K8S_OWN_POD_NAME).$(K8S_LWS_NAME).$(K8S_OWN_NAMESPACE).svc.cluster.local",  # this needs to match what's given as --own-address in the vllm start command
+            "NCCL_SOCKET_IFNAME": "eth0",
+            "GLOO_SOCKET_IFNAME": "eth0",  # maybe don't need
+            "NCCL_DEBUG": "INFO",  # TODO remove once fully tested, will keep around for now
+            "VLLM_LOGGING_LEVEL": "INFO",  # TODO remove once fully tested, will keep around for now
+            "RAY_CLUSTER_SIZE": "$(K8S_LWS_CLUSTER_SIZE)",
+        }
+
+        leader_env = common_vllm_envs.copy()
+        worker_env = common_vllm_envs.copy()
```

Comment:
are there any difference in worker vs leader env?

Reply:
there aren't, this is likely an artifact of a long time ago where I thought there was a chance the env values could be different

Reply:
got rid of leader/worker_env anyways lol